src/peerLedger.js

```diff
@@ -0,0 +1,263 @@
+const EventEmitter = require('eventemitter2')
+const BigNumber = require('big-number')
+const sha256 = require('./sha256')
+
+class NamedError extends Error {
+  constructor (msg) {
+    super(msg)
+    this.name = msg.split(' ').map(str => {
+      return str[0].toUpperCase() + str.substring(1)
+    }).join('') + 'Error'
+  }
+}
+
+function invalidDate (d) {
+  // from https://stackoverflow.com/a/1353711/680454
+  if (Object.prototype.toString.call(d) === '[object Date]') {
```

Comment:
That stackoverflow answer explains that the [object Date] was used there because instanceof breaks across frames, but we don't deal with frames here, so I'm fine with the nicer-looking instanceof option.